Add failure logs to the artifacts if CodeQL fails

jasonleenaylor opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/thecombine/1292)
<!-- Reviewable:end -->

imnasnainaec reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/1292)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @jasonleenaylor)

___
*[.github/workflows/backend.yml, line 53 at r1](https://reviewable.io/reviews/sillsdev/thecombine/1292#-MgMVqARESntGzItRni2:-MgMVqARESntGzItRni3:b-m1cln5) ([raw file](https://github.com/sillsdev/thecombine/blob/bd08f97ffc18c0470d4b9c7014cd48ae2f2bce4f/.github/workflows/backend.yml#L53)):*
> ```
>         if: ${{ failure() }}
> ```
Why the `${{ .. }}` here? The example in https://github.com/actions/upload-artifact#conditional-artifact-upload just has `  if: failure()`.


<!-- Sent from Reviewable.io -->

jasonleenaylor commented:
___
*[.github/workflows/backend.yml, line 53 at r1](https://reviewable.io/reviews/sillsdev/thecombine/1292#-MgMVqARESntGzItRni2:-MgMXa9G6gUBOn9jPo3A:b-tfdib7) ([raw file](https://github.com/sillsdev/thecombine/blob/bd08f97ffc18c0470d4b9c7014cd48ae2f2bce4f/.github/workflows/backend.yml#L53)):*
<details><summary><i>Previously, imnasnainaec (D. Ror.) wrote…</i></summary><blockquote>

> ```
>         if: ${{ failure() }}
> ```
Why the `${{ .. }}` here? The example in https://github.com/actions/upload-artifact#conditional-artifact-upload just has `  if: failure()`.
</blockquote></details>

Well, I'm not sure myself. The CodeQl team member suggested it with the `${{ }}` in one of my earlier commits I had switched it because I saw something similar, but I then saw github documentation that testing a step failure with the braces so I switched it back. :man_shrugging: 


<!-- Sent from Reviewable.io -->

imnasnainaec reviewed (approved):
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/1292)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jasonleenaylor)



<!-- Sent from Reviewable.io -->
